src/SmartTransactionsController.test.ts

```diff
@@ -343,7 +343,7 @@ const defaultState = {
 };
 
 const mockProvider = {
-  sendAsync: jest.fn(),
+  request: jest.fn(),
 };
```

Comment:
Should `mockProvider` be substituted with `getFakeProvider()` as well, or as an alternative use `mockProvider` instead of `getFakeProvider()` in the other places?

Reply:
If `mockProvider` is enough for what we need, it would probably keep tests simpler to understand